spring/src/main/java/io/cucumber/spring/SpringFactory.java

```diff
@@ -249,8 +251,10 @@ private ConfigurableApplicationContext getContext() {
 
         private void registerGlueCodeScope(ConfigurableApplicationContext context) {
             do {
-                context.getBeanFactory().registerScope(SCOPE_CUCUMBER_GLUE, new GlueCodeScope());
-                context = (ConfigurableApplicationContext) context.getParent();
+                synchronized (context) {
```

Comment:
I don't think this works. We're swapping out `context` with a different instance.  We should probably use a constant as a monitor.

Reply:
the synchronization will be on the first reference to context passed in, but I agree it looks "unclean"